index.js

```diff
@@ -0,0 +1,346 @@
+'use strict'
+
+const path = require('path')
+const fs = require('fs')
+const async = require('async')
+const { Worker } = require('worker_threads')
+const os = require('os')
+
+const Database = require('better-sqlite3')
+const Base = require('bfx-facs-base')
+
+const DB_WORKER_ACTIONS = require(
+  './worker/db-worker-actions/db-worker-actions.const'
+)
+
+class Sqlite extends Base {
+  constructor (caller, opts, ctx) {
+    super(caller, opts, ctx)
+
+    this._hasConf = false
+    this._workers = new Map()
+    this._queue = []
+
+    this.name = 'db-sqlite'
+    this.opts = this._getOpts()
+
+    this.init()
+  }
+
+  _getOpts () {
+    const {
+      unacceptableWalFileSize = 100000000,
+      minWorkersCount = 4,
+      maxWorkersCount = 16,
+      workerPathAbsolute,
+      dbPathAbsolute,
+      name,
+      label
+    } = this.opts
+
+    const _name = name ? `_${name}` : ''
+    const _label = label ? `_${label}` : ''
+    const baseName = `${this.name}${_name}${_label}.db`
+
+    const dbPath = (
+      typeof dbPathAbsolute === 'string' &&
+      path.isAbsolute(dbPathAbsolute)
+    )
+      ? path.join(dbPathAbsolute, baseName)
+      : path.join(this.caller.ctx.root, 'db', baseName)
+    const workerPath = (
+      typeof workerPathAbsolute === 'string' &&
+      path.isAbsolute(workerPathAbsolute)
+    )
+      ? workerPathAbsolute
+      : path.join(__dirname, 'worker', 'index.js')
+
+    return {
+      ...this.opts,
+      dbPath,
+      workerPath,
+      minWorkersCount,
+      maxWorkersCount,
+      unacceptableWalFileSize
+    }
+  }
+
+  _initDb (cb) {
+    const {
+      isSqliteStoredInMemory,
+      isNotWorkerSpawned,
+      workerPath,
+      dbPath: _dbPath,
+      readonly = false,
+      fileMustExist = false,
+      timeout = 5000,
+      verbose = false
+    } = this.opts
+    const params = {
+      readonly,
+      fileMustExist,
+      timeout,
+      ...(verbose ? { verbose: console.log } : {})
+    }
+    const dbPath = isSqliteStoredInMemory
+      ? ':memory:'
+      : _dbPath
+
+    try {
+      this.db = new Database(dbPath, params)
+
+      if (isNotWorkerSpawned) {
+        cb()
+
+        return
+      }
+
+      this._spawnWorkers(
+        workerPath,
+        {
+          ...params,
+          dbPath,
+          verbose
+        }
+      )
+    } catch (err) {
+      cb(err)
+    }
+
+    cb()
+  }
+
+  _start (cb) {
+    async.series([
+      (next) => { super._start(next) },
+      (next) => {
+        const {
+          isSqliteStoredInMemory,
+          dbPath
+        } = this.opts
+
+        if (isSqliteStoredInMemory) {
+          this._initDb(next)
+
+          return
+        }
+
+        const dbDir = path.dirname(dbPath)
+
+        fs.access(dbDir, fs.constants.W_OK, (err) => {
+          if (err && err.code === 'ENOENT') {
+            const msg = `the directory ${dbDir} does not exist, please create`
+
+            return next(new Error(msg))
```

Comment:
`ERR_NO_DIRECTORY`

Reply:
sure)
just wanted to do the error message similar message such as it did in [bfx-facs-db-sqlite](https://github.com/bitfinexcom/bfx-facs-db-sqlite/blob/master/index.js#L47)